.github/workflows/test-and-lint.yml

```diff
@@ -0,0 +1,43 @@
+name: Test and Lint
+
+on:
```

Comment:
I'd recommend just test and lint on all PR. We are doing public repo so it doesn't cost us anything.

Reply:
but it is running test on all PR to main though?

In what scenario will we have PRs not merging into main?

Reply:
Hmm true. I don't think we are going to ever do development branch anyway. No staging environment. Can always amend it later.